src/libhirte/common/config.h

```diff
@@ -12,3 +23,120 @@
 #define CFG_MANAGER_PORT "Port"
 #define CFG_NODE_NAME "Name"
 #define CFG_ALLOWED_NODE_NAMES "Nodes"
+
+/*
+ * An item in a configuration map
+ */
+struct config_option {
+        char *section;
+        char *name;
+        char *value;
+};
+
+/*
+ * Initialize the application configuration object.
+ *
+ * Returns 0 if successful, otherwise standard error code
+ */
+int cfg_initialize(struct hashmap **config);
```

Comment:
I would probably add a custom struct to wrap the hashmap, like:

In header: 
```
typedef struct config config;
config *cfg_new(void);
void cfg_free(config *config)
```

And then in the C file:
```
struct config {
  struct hashmap *;
};
```
This way you hide the hashmap details from the users.

Reply:
And make it later to add more fields.

Reply:
Done